pkg/events/newentry/new_entry.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package newentry
+
+import (
+	"strings"
+	"time"
+
+	"github.com/sigstore/rekor/pkg/events"
+	"golang.org/x/exp/slices"
+
+	rekor_pb "github.com/sigstore/protobuf-specs/gen/pb-go/rekor/v1"
+)
+
+const (
+	Name   = "dev.sigstore.rekor.events.v1.NewEntry"
+	Source = "/createLogEntry"
+)
+
+var ty *events.EventType
+
+func init() {
+	empty := &rekor_pb.TransparencyLogEntry{}
+	ty = events.RegisterType(Name, Source, empty.ProtoReflect().Descriptor())
+}
+
+func New(id string, entry *rekor_pb.TransparencyLogEntry, subjects []string) (*events.Event, error) {
+	slices.Sort(subjects) // Must be sorted for consistency.
+	attrs := map[string]any{
+		"time":                   time.Unix(entry.GetIntegratedTime(), 0),
+		"rekor_entry_kind":       entry.GetKindVersion().GetKind(),
+		"rekor_signing_subjects": strings.Join(subjects, ","),
```

Comment:
Is there any concept of string arrays, so that clients don't have to know subject delimiters?

Reply:
Hm, this might be an issue for filtering then, because there's no filter for "contains", only "hasPrefix", at least for GCP - https://cloud.google.com/pubsub/docs/subscription-message-filter#filtering_syntax

The good news is that for the common use case, a certificate should only contain a single subject. I found an issue tracking support for regex in filters, but there's been no progress for awhile, so this might just have to be a documented limitation unless you have any other suggestions.

Reply:
Yeah, it is an unfortunate limitation. I hope they can add `contains` at some point given that the max length of an attribute is 1024 bytes https://cloud.google.com/pubsub/quotas#resource_limits.